### PR TITLE
Update role and add MEM repo

### PR DESCRIPTION
**Pull request by saptaktakalkar**

### CODE REVIEW

**Comment by saptaktakalkar** [comment]
@sibradzic Please review and merge.

File: README.md [hunk 1]
```diff
@@ -11,7 +11,12 @@ Any pre-requisites that may not be covered by Ansible itself or the role should
 Role Variables
 --------------
 
-A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+midonet_version: "midonet-5.2"
```
Comment (sibradzic) [inline README.md]: do not hardcode midonet versions

File: README.md [hunk 2]
```diff
@@ -11,7 +11,12 @@ Any pre-requisites that may not be covered by Ansible itself or the role should
...
+midonet_release: "stable"
+openstack_version: "mitaka"
+plugin_release: "stable"
```
Comment (sibradzic) [inline README.md]: do not hardcode MidoNet & OpenStack versions, as well as branches, but parametrize them all. Te above hard-coded values should only be a default values of these parameters.

File: README.md [hunk 3]
```diff
@@ -11,7 +11,12 @@ Any pre-requisites that may not be covered by Ansible itself or the role should
...
+mem_username: # Provide MEM username
+mem_password: # Provide MEM password
```
Comment (sibradzic) [inline README.md]: Parametrize these, add some dummy defaults like 'the-user' and 'super-secret'

File: README.md (hunk 2 above)
Comment (saptaktakalkar) [inline README.md:17]: This is only shown as an example since it's a README.

File: README.md [hunk 4]
```diff
@@ -25,7 +30,13 @@ Including an example of how to use your role (for instance, with variables passe
...
     - hosts: servers
       roles:
-         - { role: username.rolename, x: 42 }
+         - role: ansible-midonet-repo
+           midonet_version: 'midonet-5.2'
+           midonet_release: 'stable'
+           openstack_version: 'mitaka'
+           plugin_release: 'stable'
+           mem_username: 'midonet'
+           mem_password: '*******'
```
Comment (sibradzic) [inline README.md]: Again, parametrize.

File: defaults/main.yml [hunk 5]
```diff
@@ -1,2 +1,12 @@
...
 # defaults file for midonet-repo
+midonet_base_url: "http://builds.midonet.org"
+midonet_version: "current"
+midonet_release: "stable"
+openstack_version: "mitaka"
+plugin_release: "stable"
+
+# TODO Hardcoded values for MEM. Update it later.
+mem_username: "midonet"
+mem_password: "*******"
```
Comment (sibradzic) [inline defaults/main.yml]: parametrize all

**Pull request merged**